### ISSUE TITLE
why Pixi is faster than other libraries?

### ISSUE THREAD

**Issue by xuMINGzhi9** [pixijs/pixijs#7736]
I want to know why Pixi is faster than other libraries?
Does Pixi do any optimization？

**Comment by ivanpopelyshev** [comment]
PixiJS is architecture.
If there's a lib with algo that is faster than something in pixi on the case you require - you can make a plugin for pixi that does the same.
I used approaches from Genome2D with pixi, I used threejs with pixi, what matters is that you can put here whatever you want.

As for default pixi algorithms:
1. sprite&mesh batching is one of fastest
2. interaction does not have any optimizations, you have to cut quad-tree from somewhere else
3. on low-level, optimization of `uniformXY` calls is the one and only, no other renderer can do that
4. there are plugins that dont exist in other libs, like `@pixi/layers` or `@pixi/graphics-smooth`.
5. default tree update (displayobjects, transforms) is slower than several private implementations (not sharing it!!!)
6. animations dont exist. You have to use `pixi-spine` which is not possible without paid `Spine` editor, or use GSAP or write your own animations that suit you

Most of our algorithms can be understood - we reject PR's that both have no docs and no understandable source code. Usually source-code is self-explanaitory.

====
In summary, PixiJS is balanced, and has superiority in certain fields. That wont help you if you ask it too heavy task without writing your own high-level algo. Some people are trying to put 1m sprites on scene without any high-level optimization, and then complain about it :)

**Comment by xuMINGzhi9** [comment]
Thanks for your reply, it worked for me.

**Comment by bigtimebuddy** [comment]
Here's @GoodBoyDigital's explanation about how batching works:

https://youtu.be/3lbDCzaLBVA?t=755

Good for basic understanding of why PixiJS is fast for simple things like Sprites.
